torch/distributed/_sharded_tensor/__init__.py

```diff
@@ -369,25 +370,19 @@ def pre_load_state_dict_hook(module, state_dict, prefix, local_metadata, strict,
                 if isinstance(state_dict[key], ShardedTensor):
                     setattr(submodule, attr_name, state_dict[key])
 
-def shard_parameter(
-        module: torch.nn.Module,
-        param_name: str,
+def shard_tensor(
```

Comment:
I think this can be a public API because the use case is kind of common? How do we decide whether to make it public or private? Out of curiosity and for my learning purpose.

Reply:
Update: after a second thought, I will make this a private API for now.

Reply:
I was thinking that this might cause confusion to the user on which API to use. But yeah this is indeed a common API, maybe we can make it public later if we find it's necessary